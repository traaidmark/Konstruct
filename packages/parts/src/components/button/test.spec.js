// PARTS [ BUTTON > TESTS ] ####################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Button from '.';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. PROPS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

const props = {
  children: 'Button Label',
  variant: 'primary',
  size: 'large',
}

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3.1. RENDER WITHOUT PROPS ...................................................

describe('Parts > Button', () => {
  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Button>{ props.children }</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// 3.1. END ....................................................................

// 3.2. RENDER WITH ALL PROPS ..................................................

describe('Parts > Button', () => {
  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Button
        variant={ props.variant }
        size={ props.size }
      >
        { props.children }
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// 3.2. END ....................................................................

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
