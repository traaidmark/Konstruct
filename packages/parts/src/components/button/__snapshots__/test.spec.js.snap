// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parts > Button renders correctly with all props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: var(--gutter-small) var(--gutter);
  background-color: var(--btn-primary--bg);
  color: var(--btn-primary--color)!important;
  font-weight: var(--weight-bold);
  font-family: var(--family-secondary);
  border: solid 2px var(--btn-primary--border);
  border-radius: var(--radius-small);
  box-shadow: none;
  -webkit-transform: translateY(0) scale(1);
  -ms-transform: translateY(0) scale(1);
  transform: translateY(0) scale(1);
  -webkit-transition: var(--transition-fast);
  transition: var(--transition-fast);
}

.c0:hover {
  color: var(--btn-primary--colorH)!important;
  background-color: var(--btn-primary--bgH);
  border-color: var(--btn-primary--borderH);
  box-shadow: 1px 1px 10px rgba(0,0,0,0.3);
  -webkit-transform: translateY(-2px) scale(1.02);
  -ms-transform: translateY(-2px) scale(1.02);
  transform: translateY(-2px) scale(1.02);
}

.c0:disabled {
  opacity: 0.5;
}

.c0:disabled:hover {
  cursor: not-allowed;
  color: var(--btn-primary--colorH)!important;
  background-color: var(--btn-primary--bg)!important;
  border-color: var(--btn-primary--border)!important;
  box-shadow: none;
  -webkit-transform: translateY(0px) scale(1);
  -ms-transform: translateY(0px) scale(1);
  transform: translateY(0px) scale(1);
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="large"
  type="button"
>
  Button Label
</button>
`;

exports[`Parts > Button renders correctly without any props 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  size="default"
  type="button"
>
  Button Label
</button>
`;
